src/main/java/net/minecraftforge/client/player/inventory/tabs/GuiTab.java

```diff
@@ -650,15 +652,9 @@ public GuiTab setTargetGui(Class<? extends GuiContainer> targetGui)
      */
     public void onTabClicked(GuiContainer guiContainer)
     {
-        int size = getTabsForGui(guiContainer.getClass()).size();
-        if (size > 0)
-        {
-            IntStream.range(0, size).filter(i ->
-                    !getTabsForGui(targetGui).contains(getTabsForGui(guiContainer.getClass()).get(i))
-            ).forEach(i ->
-                    getTabsForGui(guiContainer.getClass()).get(i).addTo(targetGui)
-            );
-        }
+        List<GuiTab> targetTabs = getTabsForGui(targetGui);
+        List<GuiTab> tabs = getTabsForGui(guiContainer.getClass());
+        tabs.stream().filter(((Predicate<GuiTab>) targetTabs::contains).negate()).forEach(t -> t.addTo(targetGui));
```

Comment:
Really, either go for the "imperative" version (explicit for loop) or at least use `Predicates.not` from Guava, the casting shenanigans are far from ideal :P

Reply:
alrighty... Its just an annoying thing that the ide suggests that it can be changed to use the "java way" :P